node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Report.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.provision.node;
+
+import com.yahoo.slime.Cursor;
+import com.yahoo.slime.Inspector;
+import com.yahoo.vespa.config.SlimeUtils;
+
+import java.time.Instant;
+
+/**
+ * A {@code Report} contains information about a node, typically found and published by host admin.
+ *
+ * <p>May serve as a base class for reports with richer information, e.g. values.
+ *
+ * @author hakonhall
+ */
+public class Report {
+    /** The time the report was created, in milliseconds since Epoch. */
+    public static final String CREATED_FIELD = "createdMillis";
+    /** The description of the error (implies wanting to fail out node). */
+    public static final String DESCRIPTION_FIELD = "description";
+
+    private final String reportId;
+    private final Instant createdTime;
+
+    // This is the serialized report which is typically richer than we know of in this class. It's up to
+    // clients of specific reports to query the details.
+    private final Inspector reportInspector;
+
+    private Report(String reportId, Instant createdTime, Inspector reportInspector) {
+        this.reportId = reportId;
+        this.createdTime = createdTime;
+        this.reportInspector = reportInspector;
+    }
+
+    /** The ID of the report. */
+    public String getReportId() {
+        return reportId;
+    }
+
+    /** The time the report was created. */
+    public Instant getCreatedTime() { return createdTime; }
+
+    /** Whether this node has severe issues that to warrant failing it. */
+    public boolean shouldFailNode() { return !getDescription().isEmpty(); }
+
+    /** A textual summary of the report. */
+    public String getDescription() { return reportInspector.field(DESCRIPTION_FIELD).asString(); }
+
+    /** For exploring the JSON (Slime) of the report. */
+    public Inspector getInspector() {
+        return reportInspector;
+    }
+
+    /** The reportInspector will be used to serialize the full report later, including any createdTime and description. */
+    public static Report fromSlime(String reportId, Inspector reportInspector) {
```

Comment:
This conflates serialisation and data model. Serialisation logic for the node object belongs in `NodeSerializer`.

Reply:
Discussed offline: Will merge this as-is for now, and consider proper modelling of individual reports later.

Reply:
Discussed offline. This report will contain an Inspector allowing clients to fetch whatever values they like from a report, and also allows an instance to loss-less serialize into JSON. Thus this class is already a semi-serializer. We'll keep serialization in mind as we move forwards.